### PR TITLE
Issue #6011: redo XML_Structure documentation section

### PR DESCRIPTION
**Pull request by romani**
this PR fixes Issue #6011

### CODE REVIEW

File: src/xdocs/config.xml [hunk 1]
```diff
@@ -952,30 +957,18 @@
...
     </section>
 
-    <section name="XML Structure">
+    <section name="XML Configuration Structure">
 
-      <b>Configuration XML Document</b>
       <p>
-        The following DTD for a configuration XML document specifies the
-        hierarchy of modules and their properties:
+        DTD is used to validate
```
Comment (rnveach) [inline src/xdocs/config.xml]: `to validate` => `to validate the`

File: src/xdocs/config.xml [hunk 2]
```diff
@@ -952,30 +957,18 @@
...
-
-&lt;!ELEMENT property EMPTY&gt;
-&lt;!ATTLIST property
-        name NMTOKEN #REQUIRED
-        value CDATA #REQUIRED
-&gt;
-      </source>
-
       <p>
         Checkstyle validates a configuration XML document when it loads
```
Comment (rnveach) [inline src/xdocs/config.xml]: `validates a configuration XML document` => `validates the configuration XML document`
`XML document` => `XML document structure`.

File: src/xdocs/config.xml [hunk 3]
```diff
@@ -952,30 +957,18 @@
 
     </section>
 
-    <section name="XML Structure">
+    <section name="XML Configuration Structure">
 
-      <b>Configuration XML Document</b>
       <p>
```
Comment (rnveach) [inline src/xdocs/config.xml]: Shouldn't this and the next paragraph be merged? They are talking about the same thing.

File: src/xdocs/config.xml [hunk 4]
```diff
@@ -952,30 +957,18 @@
...
-&lt;!ATTLIST property
-        name NMTOKEN #REQUIRED
-        value CDATA #REQUIRED
-&gt;
-      </source>
-
       <p>
         Checkstyle validates a configuration XML document when it loads
-        the document.  To validate against the above configuration DTD,
+        the document. To validate against
```
Comment (rnveach) [inline src/xdocs/config.xml]: `To validate against` => `To validate against the`

File: src/xdocs/config.xml [hunk 5]
```diff
@@ -988,44 +981,31 @@
...
       </p>
 
       <p>
         For a complete example of a configuration XML document, examine
-        file <code>docs/sun_checks.xml</code> that checks the Sun coding
-        conventions.
+        files
+        <a href=
+          "https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/sun_checks.xml">
+          sun_checks.xml
```
Comment (rnveach) [inline src/xdocs/config.xml]: Shouldn't we use a similar search as on modules links so if we move the file we won't have to remember to update the links?
https://github.com/search?q=path%3Asrc%2Fmain%2Fresources+filename%3Asun_checks.xml+repo%3Acheckstyle%2Fcheckstyle

File: src/xdocs/config.xml [hunk 6]
```diff
@@ -988,44 +981,31 @@
...
+        </a>
+        and
+        <a href=
+          "https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml">
+          google_checks.xml
```
Comment (rnveach) [inline src/xdocs/config.xml]: Same.

File: src/xdocs/config.xml [hunk 7]
```diff
@@ -912,6 +912,11 @@
         parent <code>package</code> element.
       </p>
 
+      <p>
+        Checkstyle also validates a package names XML document when it
```
[the remark this answers is not in the document]
Comment (romani) [reply src/xdocs/config.xml]: done

File: src/xdocs/config.xml [hunk 8]
```diff
@@ -952,30 +957,18 @@
 
     </section>
 
-    <section name="XML Structure">
+    <section name="XML Configuration Structure">
```
[the remark this answers is not in the document]
Comment (romani) [reply src/xdocs/config.xml]: done

File: src/xdocs/config.xml (hunk 1 above)
Comment (romani) [reply src/xdocs/config.xml]: done

File: src/xdocs/config.xml (hunk 2 above)
Comment (romani) [reply src/xdocs/config.xml]: done

File: src/xdocs/config.xml (hunk 3 above)
Comment (romani) [reply src/xdocs/config.xml]: first paragraph was removed.

File: src/xdocs/config.xml (hunk 4 above)
Comment (romani) [reply src/xdocs/config.xml]: done

File: src/xdocs/config.xml (hunk 5 above)
Comment (romani) [reply src/xdocs/config.xml]: Direct link is more user friendly, I saw a cases when users do not understand how to get to file in such search result.
Change that we will change location is not big.
We used such searches only for content in file, that is very reasonable as as content it changing and direct link to line number will not work.

File: src/xdocs/config.xml (hunk 6 above)
Comment (romani) [reply src/xdocs/config.xml]: see above

File: src/xdocs/config.xml (hunk 2 above)
Comment (rnveach) [reply src/xdocs/config.xml]: @romani Same here.

File: src/xdocs/config.xml (hunk 4 above)
Comment (rnveach) [reply src/xdocs/config.xml]: @romani same here.

File: src/xdocs/config.xml (hunk 1 above)
Comment (romani) [reply src/xdocs/config.xml]: strange. I did it. 
done.
verified after push to remote

File: src/xdocs/config.xml (hunk 2 above)
Comment (romani) [reply src/xdocs/config.xml]: strange. I did it. 
done.
verified after push to remote

File: src/xdocs/config.xml (hunk 4 above)
Comment (romani) [reply src/xdocs/config.xml]: strange. I did it. 
done.
verified after push to remote

**Comment by rnveach** [comment]
I lost power yesterday and my PC didn't auto-restart.
I thought jenkins would switch to another slave if one was down?
My slave is running now.

**Pull request merged**